src/core/stdcpp/vector.d

```diff
@@ -587,7 +587,7 @@ extern(D):
                 for (size_t i = _Size; i > 0; )
                 {
                     --i;
-                    _Get_data()._Myfirst[i].moveEmplace(_Newvec[i]);
+                    moveEmplace(_Get_data()._Myfirst[i], _Newvec[i]);
```

Comment:
O_o ... what's going on here? If that fixes a bug, then I'm worried there's an issue with the language...

Reply:
Based on your recent comment, I think this fix is hiding the issue, and you should revert this change and fix the issue instead.

Reply:
That would fix this specific case but a related problem could still arise for any struct with a public member named `moveEmplace`.

That might not be a problem. It could be a member function with a contract compatible with `core.lifetime.moveEmplace` which may have legitimate reason to exist and using it instead of the expected function could be beneficial. Against this I'll note that `move` and `core_emplace` aren't used inside `vector` with UFCS. That style was only used with that single call to `moveEmplace`.

Please confirm you have read this and still want to retain UFCS here. If so I will change the PR as requested.

Reply:
> could still arise for any struct with a public member named `moveEmplace`

Oh yeah, of course >_<

Perhaps both fixes are necessary... although it depends on if it's decided that creating a `moveEmplace` local to the object as a global override is a desired feature or not? Should it be considered that an object's local version is an intent to override the global? I think that needs to be resolved broadly.

Sadly, `moveEmplace` is a gross hack in D, and I'm not sure I would trust anyone to write it, but this pattern should should be decided and applied broadly.

I'm quite certain that it's improper for vector to have those local imports at member-scope, but the fix to this line can be seen from either perspective as you've shown :/

Reply:
Updated the PR to apply both fixes.